src/mqueryfront/src/auth/AuthPage.js

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from "react";
+import ErrorBoundary from "../components/ErrorBoundary";
+import axios from "axios";
+
+class AuthPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            error: null,
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.config == prevProps.config || !this.props.config) {
+            return;
+        }
+        if (!this.props.config["openid_auth_url"]) {
+            this.setState({ error: "OIDC not configured" });
+        }
+        const queryString = window.location.search;
+        const urlParams = new URLSearchParams(queryString);
+        const code = urlParams.get("code");
+        const payload = `grant_type=authorization_code&code=${code}&client_id=app`;
```

Comment:
Kinda hacky tbh, we assume that we don't urlencode `code`

Reply:
I agree, we can pass that payload via regular body or `params`>

Reply:
I think I've copied that code from example on the internet :seal:. It's the Internet's fault!

OTOH it's unusual today for API to get structured urlencoded POST payloads (instead of JSON) so I get where people are coming from.

Reply:
(fixed in the new version)